pkg/reconciler/revision/revision_test.go

```diff
@@ -488,118 +489,62 @@ func TestNoQueueSidecarImageUpdateFail(t *testing.T) {
-			},
-		},
-		callback: func(t *testing.T) func(runtime.Object) HookResult {
-			return func(obj runtime.Object) HookResult {
-				revision := obj.(*v1.Revision)
-				t.Logf("Revision updated: %v", revision.Name)
-
-				expected := int64(3)
-				got := *(revision.Spec.ContainerConcurrency)
-				if got != expected {
-					return HookComplete
-				}
-
-				t.Logf("No update occurred; expected: %d got: %d", expected, got)
-				return HookIncomplete
-			}
-		},
-	}}
+	ctx, informers, ctrl, watcher := newTestControllerWithConfig(t, nil)
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			ctx, informers, ctrl, watcher := newTestControllerWithConfig(t, nil)
-
-			ctx, cancel := context.WithCancel(ctx)
-			grp := errgroup.Group{}
-
-			servingClient := fakeservingclient.Get(ctx)
-
-			rev := testRevision(getPodSpec())
-			revClient := servingClient.ServingV1().Revisions(rev.Namespace)
-
-			h := NewHooks()
+	ctx, cancel := context.WithCancel(ctx)
+	grp := errgroup.Group{}
 
-			h.OnUpdate(&servingClient.Fake, "revisions", test.callback(t))
+	servingClient := fakeservingclient.Get(ctx)
 
-			waitInformers, err := controller.RunInformers(ctx.Done(), informers...)
-			if err != nil {
-				t.Fatal("Failed to start informers:", err)
-			}
-			defer func() {
-				cancel()
-				if err := grp.Wait(); err != nil {
-					t.Errorf("Wait() = %v", err)
-				}
-				waitInformers()
-			}()
+	rev := testRevision(getPodSpec())
 
-			if err := watcher.Start(ctx.Done()); err != nil {
-				t.Fatal("Failed to start configuration manager:", err)
-			}
+	waitInformers, err := controller.RunInformers(ctx.Done(), informers...)
+	if err != nil {
+		t.Fatal("Failed to start informers:", err)
+	}
+	defer func() {
+		cancel()
+		if err := grp.Wait(); err != nil {
+			t.Errorf("Wait() = %v", err)
+		}
+		waitInformers()
+	}()
 
-			grp.Go(func() error { return ctrl.Run(1, ctx.Done()) })
+	if err := watcher.Start(ctx.Done()); err != nil {
+		t.Fatal("Failed to start configuration manager:", err)
+	}
 
-			revClient.Create(rev)
-			revL := fakerevisioninformer.Get(ctx).Lister()
-			if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
-				l, err := revL.List(labels.Everything())
-				// We only create a single revision.
-				return len(l) > 0, err
-			}); err != nil {
-				t.Fatal("Failed to see Revision propagation:", err)
-			}
-			t.Log("Seen revision propagation")
+	grp.Go(func() error { return ctrl.Run(1, ctx.Done()) })
+
+	servingClient.ServingV1().Revisions(rev.Namespace).Create(rev)
+	revL := fakerevisioninformer.Get(ctx).Lister()
+	if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		// The only error we're getting in the test reasonably is NotFound.
+		r, _ := revL.Revisions(rev.Namespace).Get(rev.Name)
+		// We only create a single revision, but make sure it is reconciled.
+		return r != nil && r.Status.ObservedGeneration == r.Generation, nil
```

Comment:
I believe this will always be true because the Generation defaults to 0. Can you make sure to set one explicitly?

Reply:
done.

Reply:
well, I check Observed is > 0.